[bug] tauri-utils-2.0.0-beta.13 throwing compile error `not found in 'schemars::_private'`

### Describe the bug

When I try to install cargo crates within the vscode devcontainer I'm hitting a `not found` error

```bash
error[E0425]: cannot find function `apply_metadata` in module `schemars::_private`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-utils-2.0.0-beta.13/src/config.rs:216:27
    |
216 |       schemars::_private::apply_metadata(
    |                           ^^^^^^^^^^^^^^ not found in `schemars::_private`

error[E0425]: cannot find function `apply_metadata` in module `schemars::_private`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-utils-2.0.0-beta.13/src/config.rs:223:27
    |
223 |       schemars::_private::apply_metadata(
    |                           ^^^^^^^^^^^^^^ not found in `schemars::_private`
```

### Reproduction

```bash
cd /path/to/app/src-tauri
cargo install --path .
```

### Expected behavior

Crate to be successfully installed without errors

### Full `tauri info` output

```text
[✔] Environment
    - OS: Debian 12 X64
    ✔ webkit2gtk-4.1: 2.42.5
    ✔ rsvg2: 2.54.7
    ✔ rustc: 1.77.2 (25ef9e3d8 2024-04-09)
    ✔ cargo: 1.77.2 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: 1.77.2-x86_64-unknown-linux-gnu (default)
    - node: 20.12.2
    - pnpm: 9.1.0
    - yarn: 1.22.19
    - npm: 10.5.0
    - bun: 1.1.7

[-] Packages
    - tauri [RUST]: 2.0.0-beta.17
    - tauri-build [RUST]: 2.0.0-beta.13
    - wry [RUST]: 0.39.3
    - tao [RUST]: 0.27.1
    - @tauri-apps/api [NPM]: 2.0.0-beta.11
    - @tauri-apps/cli [NPM]: 2.0.0-beta.15

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:5173/
    - framework: Svelte
    - bundler: Vite
```


### Stack trace

```text
$ cargo install --path .
  Installing app v0.0.0 (/workspaces/app/src-tauri)
    Updating crates.io index
   Compiling proc-macro2 v1.0.81
   Compiling unicode-ident v1.0.12
   Compiling serde v1.0.200
   Compiling hashbrown v0.14.5
   Compiling equivalent v1.0.1
   Compiling winnow v0.5.40
   Compiling smallvec v1.13.2
   Compiling autocfg v1.3.0
   Compiling heck v0.5.0
   Compiling libc v0.2.154
   Compiling cfg-if v1.0.0
   Compiling syn v1.0.109
   Compiling pkg-config v0.3.30
   Compiling target-lexicon v0.12.14
   Compiling version-compare v0.2.0
   Compiling siphasher v0.3.11
   Compiling ppv-lite86 v0.2.17
   Compiling thiserror v1.0.59
   Compiling getrandom v0.1.16
   Compiling indexmap v2.2.6
   Compiling pin-project-lite v0.2.14
   Compiling lock_api v0.4.12
   Compiling parking_lot_core v0.9.10
   Compiling version_check v0.9.4
   Compiling once_cell v1.19.0
   Compiling phf_shared v0.8.0
   Compiling phf_shared v0.10.0
   Compiling semver v1.0.22
   Compiling futures-core v0.3.30
   Compiling proc-macro-hack v0.5.20+deprecated
   Compiling cfg-expr v0.15.8
   Compiling fnv v1.0.7
   Compiling convert_case v0.4.0
   Compiling futures-io v0.3.30
   Compiling ident_case v1.0.1
   Compiling strsim v0.10.0
   Compiling memchr v2.7.2
   Compiling phf_shared v0.11.2
   Compiling getrandom v0.2.15
   Compiling slab v0.4.9
   Compiling tinyvec_macros v0.1.1
   Compiling value-bag v1.9.0
   Compiling rand_core v0.6.4
   Compiling rand_core v0.5.1
   Compiling scopeguard v1.2.0
   Compiling quote v1.0.36
   Compiling new_debug_unreachable v1.0.6
   Compiling rand_chacha v0.3.1
   Compiling rand_pcg v0.2.1
   Compiling rand_chacha v0.2.2
   Compiling byteorder v1.5.0
   Compiling syn v2.0.60
   Compiling rand v0.8.5
   Compiling rand v0.7.3
   Compiling aho-corasick v1.1.3
   Compiling tinyvec v1.6.0
   Compiling regex-syntax v0.8.3
   Compiling serde_json v1.0.116
   Compiling mac v0.1.1
   Compiling precomputed-hash v0.1.1
   Compiling futf v0.1.5
   Compiling parking_lot v0.12.2
   Compiling indexmap v1.9.3
   Compiling phf_generator v0.8.0
   Compiling phf_generator v0.10.0
   Compiling phf_generator v0.11.2
   Compiling phf_codegen v0.8.0
   Compiling itoa v1.0.11
   Compiling string_cache_codegen v0.5.2
   Compiling phf_codegen v0.10.0
   Compiling percent-encoding v2.3.1
   Compiling selectors v0.22.0
   Compiling ryu v1.0.17
   Compiling log v0.4.21
   Compiling utf-8 v0.7.6
   Compiling unicode-normalization v0.1.23
   Compiling dtoa v1.0.9
   Compiling unicode-bidi v0.3.15
   Compiling bitflags v1.3.2
   Compiling futures-sink v0.3.30
   Compiling tendril v0.4.3
   Compiling form_urlencoded v1.2.1
   Compiling dtoa-short v0.3.4
   Compiling phf v0.10.1
   Compiling itoa v0.4.8
   Compiling markup5ever v0.11.0
   Compiling hashbrown v0.12.3
   Compiling camino v1.1.6
   Compiling nodrop v0.1.14
   Compiling stable_deref_trait v1.2.0
   Compiling unic-char-range v0.9.0
   Compiling futures-task v0.3.30
   Compiling alloc-no-stdlib v2.0.4
   Compiling idna v0.5.0
   Compiling matches v0.1.10
   Compiling pin-utils v0.1.0
   Compiling unic-common v0.9.0
   Compiling unic-char-property v0.9.0
   Compiling alloc-stdlib v0.2.2
   Compiling unic-ucd-version v0.9.0
   Compiling servo_arc v0.1.1
   Compiling fxhash v0.2.1
   Compiling uuid v1.8.0
   Compiling schemars v0.8.18
   Compiling thin-slice v0.1.1
   Compiling unic-ucd-ident v0.9.0
   Compiling brotli-decompressor v2.5.1
   Compiling regex-automata v0.4.6
   Compiling same-file v1.0.6
   Compiling anyhow v1.0.82
   Compiling dyn-clone v1.0.17
   Compiling cfb v0.7.3
   Compiling walkdir v2.5.0
   Compiling dunce v1.0.4
   Compiling glob v0.3.1
   Compiling cc v1.0.97
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling futures-channel v0.3.30
   Compiling crossbeam-utils v0.8.19
   Compiling infer v0.15.0
   Compiling brotli v3.5.0
   Compiling darling_core v0.20.8
   Compiling serde_derive_internals v0.29.0
   Compiling heck v0.4.1
   Compiling typenum v1.17.0
   Compiling generic-array v0.14.7
   Compiling gio v0.18.4
   Compiling dirs-sys-next v0.1.2
   Compiling memoffset v0.9.1
   Compiling dirs-next v2.0.0
   Compiling tracing-core v0.1.32
   Compiling regex v1.10.4
   Compiling socket2 v0.5.7
   Compiling num_cpus v1.16.0
   Compiling ring v0.17.8
   Compiling gtk v0.18.1
   Compiling x11 v2.21.0
   Compiling serde_derive v1.0.200
   Compiling thiserror-impl v1.0.59
   Compiling cssparser-macros v0.6.1
   Compiling phf_macros v0.11.2
   Compiling ctor v0.2.8
   Compiling futures-macro v0.3.30
   Compiling darling_macro v0.20.8
   Compiling schemars_derive v0.8.18
   Compiling cssparser v0.27.2
   Compiling phf_macros v0.8.0
   Compiling html5ever v0.26.0
   Compiling derive_more v0.99.17
   Compiling darling v0.20.8
   Compiling phf v0.11.2
   Compiling serde_with_macros v3.8.1
   Compiling tracing-attributes v0.1.27
   Compiling futures-util v0.3.30
   Compiling phf v0.8.0
   Compiling crc32fast v1.4.0
   Compiling simd-adler32 v0.3.7
   Compiling rustls-pki-types v1.5.0
   Compiling untrusted v0.9.0
   Compiling httparse v1.8.0
   Compiling spin v0.9.8
   Compiling adler v1.0.2
   Compiling parking v2.2.0
   Compiling miniz_oxide v0.7.2
   Compiling pin-project-internal v1.1.5
   Compiling concurrent-queue v2.5.0
   Compiling crossbeam-channel v0.5.12
   Compiling futures-executor v0.3.30
   Compiling x11-dl v2.21.0
   Compiling try-lock v0.2.5
   Compiling rustls v0.22.4
   Compiling raw-window-handle v0.6.1
   Compiling pin-project v1.1.5
   Compiling want v0.3.1
   Compiling flate2 v1.0.30
   Compiling block-buffer v0.10.4
   Compiling crypto-common v0.1.6
   Compiling fdeflate v0.3.4
   Compiling serde_spanned v0.6.5
   Compiling toml_datetime v0.6.3
   Compiling string_cache v0.8.7
   Compiling toml_edit v0.20.2
   Compiling toml_edit v0.19.15
   Compiling url v2.5.0
   Compiling rustc_version v0.4.0
   Compiling toml v0.8.2
   Compiling cargo-platform v0.1.8
   Compiling urlpattern v0.2.0
   Compiling system-deps v6.2.2
   Compiling kuchikiki v0.8.2
   Compiling cargo_metadata v0.18.1
   Compiling json-patch v1.4.0
   Compiling bitflags v2.5.0
   Compiling serde_with v3.8.1
   Compiling proc-macro-crate v2.0.2
   Compiling glib-sys v0.18.1
   Compiling gobject-sys v0.18.0
   Compiling gio-sys v0.18.1
   Compiling gdk-sys v0.18.0
   Compiling pango-sys v0.18.0
   Compiling atk-sys v0.18.0
   Compiling cairo-sys-rs v0.18.2
   Compiling gdk-pixbuf-sys v0.18.0
   Compiling tauri-utils v2.0.0-beta.13
   Compiling glib-macros v0.18.5
   Compiling gtk-sys v0.18.0
error[E0425]: cannot find function `apply_metadata` in module `schemars::_private`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-utils-2.0.0-beta.13/src/config.rs:216:27
    |
216 |       schemars::_private::apply_metadata(
    |                           ^^^^^^^^^^^^^^ not found in `schemars::_private`

error[E0425]: cannot find function `apply_metadata` in module `schemars::_private`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-utils-2.0.0-beta.13/src/config.rs:223:27
    |
223 |       schemars::_private::apply_metadata(
    |                           ^^^^^^^^^^^^^^ not found in `schemars::_private`

   Compiling bytes v1.6.0
   Compiling embed-resource v2.4.2
   Compiling toml v0.7.8
   Compiling proc-macro-crate v1.3.1
For more information about this error, try `rustc --explain E0425`.
error: could not compile `tauri-utils` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `app v0.0.0 (/workspaces/app/src-tauri)`, intermediate artifacts can be found at `/workspaces/app/src-tauri/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```


### Additional context

_No response_

--- a/.changes/fix-schemars-compatibility.md
+++ b/.changes/fix-schemars-compatibility.md
@@ -1,0 +1,5 @@
+---
+tauri-utils: "patch:bug"
+---
+
+Fixes `schemars` compilation issue.

--- a/core/tauri-build/Cargo.toml
+++ b/core/tauri-build/Cargo.toml
@@ -41,7 +41,7 @@ semver = "1"
 dirs-next = "2"
 glob = "0.3"
 toml = "0.8"
-schemars = { version = "0.8", features = [ "preserve_order" ] }
+schemars = { version = "0.8.18", features = [ "preserve_order" ] }
 
 [features]
 default = [ "config-json" ]

--- a/core/tauri-config-schema/Cargo.toml
+++ b/core/tauri-config-schema/Cargo.toml
@@ -6,7 +6,7 @@ publish = false
 
 [build-dependencies]
 tauri-utils = { features = [ "schema" ], path = "../tauri-utils" }
-schemars = { version = "0.8", features = ["url", "preserve_order"] }
+schemars = { version = "0.8.18", features = ["url", "preserve_order"] }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 url = { version = "2.3", features = ["serde"] }

--- a/core/tauri-plugin/Cargo.toml
+++ b/core/tauri-plugin/Cargo.toml
@@ -34,7 +34,7 @@ tauri-utils = { version = "2.0.0-beta.13", default-features = false, features = 
 serde_json = { version = "1", optional = true }
 glob = { version = "0.3", optional = true }
 toml = { version = "0.8", optional = true }
-schemars = { version = "0.8", features = [ "preserve_order" ] }
+schemars = { version = "0.8.18", features = [ "preserve_order" ] }
 walkdir = { version = "2", optional = true }
 
 [target."cfg(target_os = \"macos\")".dependencies]

--- a/core/tauri-utils/Cargo.toml
+++ b/core/tauri-utils/Cargo.toml
@@ -23,7 +23,7 @@ html5ever = "0.26"
 kuchiki = { package = "kuchikiki", version = "0.8" }
 proc-macro2 = { version = "1", optional = true }
 quote = { version = "1", optional = true }
-schemars = { version = "0.8", features = [ "url" ], optional = true }
+schemars = { version = "0.8.18", features = [ "url" ], optional = true }
 serde_with = "3"
 aes-gcm = { version = "0.10", optional = true }
 getrandom = { version = "0.2", optional = true, features = [ "std" ] }

--- a/core/tauri-utils/src/config.rs
+++ b/core/tauri-utils/src/config.rs
@@ -213,19 +213,13 @@ impl schemars::JsonSchema for BundleTarget {
         ..Default::default()
       }
       .into(),
-      schemars::_private::apply_metadata(
+      schemars::_private::metadata::add_description(
         gen.subschema_for::<Vec<BundleType>>(),
-        schemars::schema::Metadata {
-          description: Some("A list of bundle targets.".to_owned()),
-          ..Default::default()
-        },
+        "A list of bundle targets.",
       ),
-      schemars::_private::apply_metadata(
+      schemars::_private::metadata::add_description(
         gen.subschema_for::<BundleType>(),
-        schemars::schema::Metadata {
-          description: Some("A single bundle target.".to_owned()),
-          ..Default::default()
-        },
+        "A single bundle target.",
       ),
     ];
 

